tests/Aspire.Hosting.Tests/AWS/AWSCloudFormationResourceTests.cs

```diff
@@ -0,0 +1,113 @@
+using Aspire.Hosting.Utils;
+using Xunit;
+
+namespace Aspire.Hosting.Tests.AWS;
+public class AWSCloudFormationResourceTests
+{
+    [Fact]
+    public void AddAWSCloudFormationStackResourceTest()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+
+        var awsSdkConfig = builder.AddAWSSDKConfig()
+                                .WithRegion(RegionEndpoint.USWest2)
+                                .WithProfile("test-profile");
+
+        var resource = builder.AddAWSCloudFormationStack("ExistingStack")
+                                                    .WithReference(awsSdkConfig)
+                                                    .Resource;
+
+        Assert.Equal("ExistingStack", resource.Name);
+        Assert.NotNull(resource.AWSSDKConfig);
+        Assert.Equal(RegionEndpoint.USWest2, resource.AWSSDKConfig.Region);
+        Assert.Equal("test-profile", resource.AWSSDKConfig.Profile);
+    }
+
+    [Fact]
+    public void AddAWSCloudFormationTemplateResourceTest()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+
+        var awsSdkConfig = builder.AddAWSSDKConfig()
+                                .WithRegion(RegionEndpoint.USWest2)
+                                .WithProfile("test-profile");
+
+        var resource = builder.AddAWSCloudFormationTemplate("NewStack", "cf.template")
+                                                    .WithParameter("key1", "value1")
+                                                    .WithParameter("key2", "value2")
+                                                    .WithReference(awsSdkConfig)
+                                                    .Resource as CloudFormationTemplateResource;
+
+        Assert.NotNull(resource);
+        Assert.Equal("NewStack", resource.Name);
+        Assert.Equal("cf.template", resource.TemplatePath);
+        Assert.NotNull(resource.AWSSDKConfig);
+        Assert.Equal(RegionEndpoint.USWest2, resource.AWSSDKConfig.Region);
+        Assert.Equal("test-profile", resource.AWSSDKConfig.Profile);
+
+        Assert.Equal(2, resource.CloudFormationParameters.Count);
+        Assert.Equal("value1", resource.CloudFormationParameters["key1"]);
+        Assert.Equal("value2", resource.CloudFormationParameters["key2"]);
+    }
+
+    [Fact]
+    public void ManifestAWSCloudFormationStackResourceTest()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+
+        var resourceBuilder = builder.AddAWSCloudFormationStack("ExistingStack");
+
+        var projectBuilder = builder.AddProject<Projects.ServiceA>("serviceA")
+                                    .WithReference(resourceBuilder);
+
+        var resource = resourceBuilder.Resource as CloudFormationStackResource;
+        Assert.NotNull(resource);
+        var obj = ManifestUtils.GetManifest(resource.WriteToManifest);
+
+        Assert.NotNull(obj);
+        Assert.Equal("aws.cloudformation.stack.v0", obj["type"]?.ToString());
```

Comment:
Take a look at how we've been [asserting these changes recently](https://github.com/dotnet/aspire/blob/493f631075a04382bef6d312945ff60a58f30975/tests/Aspire.Hosting.Tests/Azure/AzureBicepResourceTests.cs#L452-L476).

It's much more thorough and catches more bugs 😄

Reply:
Thanks, changed to that style.